### PR TITLE
[ADD] crm_l10n_eu_nace: NACE for leads

### PR DESCRIPTION
**Pull request by yajo**
This module adds the concept of NACE activity to the CRM main objects (lead,
opportunity), and propagates them through the workflow.

Depends on:

- [x] https://github.com/OCA/community-data-files/pull/53
- [x] https://github.com/OCA/community-data-files/pull/90

@Tecnativa TT17645

### CODE REVIEW

File: crm_l10n_eu_nace/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+* Pedro M. Baeza (`Tecnativa <https://www.tecnativa.com/>`__)
+* Antonio Espinosa (`Tecnativa <https://www.tecnativa.com/>`__)
+* David Vidal (`Tecnativa <https://www.tecnativa.com/>`__)
+* Jairo Llopis (`Tecnativa <https://www.tecnativa.com/>`__)
```
Comment (pedrobaeza) [inline crm_l10n_eu_nace/readme/CONTRIBUTORS.rst]: Better to use 2 indentation levels as we do in other modules
```suggestion
* `Tecnativa <https://www.tecnativa.com/>`__:

  * Pedro M. Baeza
  * Antonio Espinosa
  * David Vidal
  * Jairo Llopis
```

File: crm_l10n_eu_nace/README.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+This module adds the concept of NACE activity to the CRM main objects (lead,
+opportunity), and propagates them through the workflow.
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Bug Tracker
```
Comment (pedrobaeza) [inline crm_l10n_eu_nace/README.rst:36]: Add `INSTALL.rst` saying that it depends on `l10n_eu_nace` module in OCA/community-data-files.

File: crm_l10n_eu_nace/models/crm_lead.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    @api.multi
+    def _create_lead_partner_data(self, *args, **kwargs):
+        """Propagate NACE activity to created partner."""
+        result = super()._create_lead_partner_data(*args, **kwargs)
+        if self.nace_id:
+            result["nace_id"] = self.nace_id.ids
+        if self.secondary_nace_ids:
+            result["secondary_nace_ids"] = [
+                (4, self.secondary_nace_ids.ids, 0)
```
Comment (pedrobaeza) [inline crm_l10n_eu_nace/models/crm_lead.py]:
```suggestion
                (4, x) for x in self.secondary_nace_ids.ids
```

File: crm_l10n_eu_nace/tests/test_crm_nace.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    def test_data_transferred_to_partner(self):
+        """Data is moved to partner when creating it from lead."""
+        # Create a lead with nace codes but without partner yet
+        lead = self.env["crm.lead"].create(
+            {
+                "name": "test lead",
+                "partner_name": "someone",
+                "nace_id": self.nace_0.id,
+                "secondary_nace_ids": [
+                    (4, (self.nace_1.id, self.nace_2.id), 0)],
```
Comment (pedrobaeza) [inline crm_l10n_eu_nace/tests/test_crm_nace.py]:
```suggestion
                    (4, self.nace_1.id), (4, self.nace_2.id)],
```

File: crm_l10n_eu_nace/tests/test_crm_nace.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            }
+        )
+        self.assertFalse(lead.partner_id)
+        # Create that partner automatically
+        lead.handle_partner_assignation()
+        self.assertEqual(lead.partner_id.nace_id, self.nace_0)
+        self.assertEqual(
+            lead.partner_id.secondary_nace_id, self.nace_1 | self.nace_2)
```
Comment (pedrobaeza) [inline crm_l10n_eu_nace/tests/test_crm_nace.py]:
```suggestion
            lead.partner_id.secondary_nace_ids, self.nace_1 | self.nace_2)
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: crm_l10n_eu_nace/models/crm_lead.py (hunk 3 above)
Comment (yajo) [reply crm_l10n_eu_nace/models/crm_lead.py]: Don't you always need the 3rd element in the tuple?

Comment (pedrobaeza) [reply crm_l10n_eu_nace/models/crm_lead.py]: Not really, but that's not the main problem here: you have to add a `(4, ...` command for each element.

File: crm_l10n_eu_nace/README.rst (hunk 2 above)
Comment (yajo) [reply crm_l10n_eu_nace/README.rst:36]: All fixed, thanks.

File: crm_l10n_eu_nace/models/crm_lead.py (hunk 3 above)
Comment (pedrobaeza) [reply crm_l10n_eu_nace/models/crm_lead.py]: This is still not changed

**Review by pedrobaeza — APPROVED** [review]

**Review by ValentinVinagre — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]
LGTM 👍

**Pull request merged**